File: cpp/LinearSearch.cpp
// Program for Linear Search using Functions 

#include<iostream>
using namespace std;

int LinearSearch(int A[],int n,int key)
{   
    int i;
    for( i=0;i<n;i++)
    {
      if(key==A[i])
      {
       cout<<"Element found at index : "<<i<<endl;
       return 0;
      }
    }
    

 
    if(i==n)
    {
    cout<<"Element not Found";
     return 0;
    }
    
}

int main()
{
 int num,k;
 cout<<"Enter the number of Elements in Array: ";
 cin>>num;
 int A[num];
 cout<<"Enter " << num << " elements of Array: ";
 
 for(int i=0;i<num;i++)
 {
    cin>>A[i];
 }
 
 cout<<"Enter an Element to be Searched: ";
 cin>>k;
 int index=LinearSearch(A,num,k);
 
 
 
 
}

File: Java/LinearSearch.java
import java.util.*;
class LinearSearch
{
    static int linearSearch(int[] arr, int n)
    {
        for(int i=0;i<arr.length;i++)
        {
            if(arr[i]==n)
                return i;
        }
        return -1;
    }
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter the size of the array");
        int size=sc.nextInt();
        System.out.println("Enter the array Elememts");
        int[] arr=new int[size];
        for(int i=0;i<size;i++)
        {
            arr[i]=sc.nextInt();
        }
        System.out.println("Enter the elements to be searched");
        int n=sc.nextInt();
        int position = linearSearch(arr, n);
        if(position!=-1)
            System.out.println("Elemen foun at index "+position);
        else
            System.out.println("Element not found");
    }
}